### PR TITLE
SN238: Add aggregation endpoints for celery async task results 

### PR DESCRIPTION
**Pull request by prakharrathi25**
# Description

This PR adds two endpoints which show the Celery TaskResults from the database queried by the task name for a particular project ID or a dataset instance. 

The endpoints are 
1. `GET /projects/<project_id>/get_async_task_results/?task_name=<task_name>` 
2. `GET /data/instances/<instance_id>/get_async_task_results/?task_name=<task_name>`

Fixes [SN-238](https://project-sunbird.atlassian.net/browse/SN-238?atlOrigin=eyJpIjoiMWQyODllNmRlOGI0NGM1ZDlhZDFkN2IwMTBjMzM1YmIiLCJwIjoiaiJ9)

image

## Type of change

Please delete options that are not relevant.

- [x] New feature (non-breaking change which adds functionality)

# How Has This Been Tested?

- [x] Tested the API endpoint on Swagger with the 'Try it out' feature
- [x] Any other tests you performed for checking edge cases.
- [x] Any other tests you performed for checking exceptions.

# Checklist:

- [x] My code follows the style guidelines of this project
- [x] I have performed a self-review of my own code
- [x] I have commented my code, particularly in hard-to-understand areas

## Checklist for API

If you added new api endpoints / made changes to exsting endpoints, please fill this checklist. 

- [x] The endpoint is accessible through Swagger.
- [x] The endpoint supports permissions and authentication for different roles.
- [x] All exceptions have been handled and appropriate status code is returned to the user.
- [x] I have added docstrings for ViewSets / Serializers.

### CODE REVIEW

File: backend/dataset/views.py [hunk 1]
```diff
@@ -366,6 +363,116 @@ def projects(self, request, pk):
...
+                "'project_id': " + "'" + str(pk) + "'" for pk in project_ids
+            ]
+
+            # Turn list of project ID keywords into list of Q objects
+            queries = [
+                Q(task_kwargs__contains=project_keyword)
+                for project_keyword in project_id_keyword_args
+            ]
+            print("The QUERY", queries)
+            query = queries.pop()
```
Comment (ishvindersethi22) [inline backend/dataset/views.py:411]: Please handle with exception when the list is empty

Comment (prakharrathi25) [reply backend/dataset/views.py]: I have updated this.

**Review by ishvindersethi22 — APPROVED** [review]
Tested on dev server.
